src/tools/netlink.c

```diff
@@ -421,6 +421,8 @@ static int nlparsemsg_address(struct ifaddrmsg *ifa, void *buf, size_t len, cJSO
 						cJSON_AddStringToObject(addr, type_str, "multicast");
 					else if(private_net(*in, false))
 						cJSON_AddStringToObject(addr, type_str, "private");
+					else if((in->s_addr & htonl(0xffc00000)) == htonl(0x64400000))
+						cJSON_AddStringToObject(addr, type_str, "shared");
```

Comment:
I wonder if "shared" is the best term here, maybe say explicitly what this is?
```suggestion
						// RFC 6598: Carrier-Grade NAT (CGN) 100.64.0.0/10
						cJSON_AddStringToObject(addr, type_str, "Carrier-Grade NAT");
```
This is entirely meant as a starting point for discussion, '"shared"' may be equally fine.

Reply:
The RFC specifies that the shared space is to be used for CGNat, I think CGN is proper.

>  This document requests the allocation of an IPv4 /10 address block to
>    be used as Shared Address Space to accommodate the needs of Carrier-
>    Grade NAT (CGN) devices.  It is anticipated that Service Providers
>    will use this Shared Address Space to number the interfaces that
>    connect CGN devices to Customer Premises Equipment (CPE).